lib/katello_attach_subscription/host_matcher.rb

```diff
@@ -1,13 +1,14 @@
 module KatelloAttachSubscription
   class HostMatcher
     def self.match_host(host, config)
+      host['type'] = KatelloAttachSubscription::FactAnalyzer.system_type(host) unless host['type']
       valid = true
       valid = false unless self.match_hostname(config['hostname'], host['name'])
       valid = false unless self.match_type(config['type'], host['type'])
       host_facts = host['facts'] || {}
-      config.fetch('match', []).each do |match|
+      config.fetch('facts', []).each do |match|
         matcher = match['matcher'] || 'string'
-        valid = false unless self.match_matcher(match['value'], host_facts[match['fact']], matcher)
+        valid = false unless self.match_matcher(match['value'], host_facts[match['name']], matcher)
```

Comment:
valid is always false, also when match_matcher return true
I suggest to change in `valid = self.match_matcher(match['value'], host_facts[match['name']], matcher)`

Reply:
`valid` starts out as `true` in line 5. afterwards it's flipped to `false` when one of the matchers fail. this is successfully tested by the `test_host_rhel_rhv_vm` test